src/protocols/secure_channel/CASESession.h

```diff
@@ -168,7 +168,44 @@ class DLL_EXPORT CASESession : public Messaging::UnsolicitedMessageHandler,
      **/
     void Clear();
 
+    void InvalidateIfPendingEstablishment();
+
 private:
+    class CASESessionFabricDelegate final : public chip::FabricTable::Delegate
+    {
+    public:
+        CASESessionFabricDelegate(CASESession * caseSession) :  mCASESession(caseSession) {}
+
+        void OnFabricDeletedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) override
+        {
+            (void) fabricTable;
+            (void) fabricIndex;
+            mCASESession->InvalidateIfPendingEstablishment();
```

Comment:
Isn't this going to over-zealously invalidate all pending sessions _even if_ they are not targeting the fabric in question? That doesn't seem desirable.

We should _only_ invalidate sessions where we know for certain they are targeting the fabric in question. In almost all cases, I believe calling `GetPeer` and matching the fabric index within that against this would be sufficient. `mPeerNodeId` only gets set when we conclusively know the peer we're interacting with. In the initiator role, this is set immediately at session establishment time. In the responder role, this is set upon validating Sigma3.

Reply:
Minor update - on the responder side, the fabric index is known by the time Sigma1 processing is complete.

Reply:
There is an alternative approach.

We can invalid the pairing via `SessionManager`, the `PairingSession` can handle `OnSessionReleased` event, if we call `MarkForRemoval` on the session object, the pairing will be terminated.

This approach shares most code with PR #19328, but requires that the ScopedNodeId is set to the session object when handling Sigma1